ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -694,7 +695,9 @@ public void alterTable(String catName, String dbName, String tblName, Table newT
       AcidUtils.TableSnapshot tableSnapshot = null;
       if (transactional) {
         if (replWriteId > 0) {
-          ValidWriteIdList writeIds = AcidUtils.getTableValidWriteIdListWithTxnList(conf, dbName, tblName);
+          ValidWriteIdList writeIds = new ValidReaderWriteIdList(TableName.getDbTable(dbName, tblName),
```

Comment:
Pls add a comment on why this hardcoding logic works for repl flow.

Reply:
Done.